core/src/components/select/select.tsx

```diff
@@ -983,13 +998,64 @@ export class Select implements ComponentInterface {
         aria-label={this.ariaLabel}
         aria-haspopup="dialog"
         aria-expanded={`${isExpanded}`}
+        aria-describedby={this.getHintTextID()}
+        aria-invalid={this.getHintTextID() === this.errorTextId}
         onFocus={this.onFocus}
         onBlur={this.onBlur}
         ref={(focusEl) => (this.focusEl = focusEl)}
       ></button>
     );
   }
 
+  private getHintTextID(): string | undefined {
+    const { el, helperText, errorText, helperTextId, errorTextId } = this;
+
+    if (el.classList.contains('ion-touched') && el.classList.contains('ion-invalid') && errorText) {
+      return errorTextId;
+    }
+
+    if (helperText) {
+      return helperTextId;
+    }
+
+    return undefined;
+  }
+
+  /**
+   * Renders the helper text or error text values
+   */
+  private renderHintText() {
+    const { helperText, errorText, helperTextId, errorTextId } = this;
+
+    return [
+      <div id={helperTextId} class="helper-text" part="supporting-text helper-text">
+        {helperText}
+      </div>,
+      <div id={errorTextId} class="error-text" part="supporting-text error-text">
+        {errorText}
+      </div>,
```

Comment:
I added the `supporting-text` part so users could change them at the same time using: 

```css
ion-select::part(supporting-text) {
  font-size: 20px;
}
```

but if we think it should only be the one part they could write this without it:

``` css
ion-select::part(helper-text),
ion-select::part(error-text) {
  font-size: 20px;
}
```

I am copying [Material Design 2](https://m2.material.io/components/text-fields#interactive-demo) and [Material Design 3](https://m3.material.io/components/text-fields/overview) naming here, but let me know if there are other ideas.